### PR TITLE
AIP-84: Migrating GET one queued asset events for DAG to fastAPI

### PR DESCRIPTION
**Pull request by amoghrajesh**
related: https://github.com/apache/airflow/issues/42370

Same setup as https://github.com/apache/airflow/pull/44124 and dependent on https://github.com/apache/airflow/pull/44124

Responses:
1. Legacy
image

2. FastAPI
image

With time filtering
1. Legacy
image

2. FastAPI
image

Swagger spec:
image

image

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by amoghrajesh** [comment]
Only last commit is relevant

File: airflow/api_fastapi/core_api/routes/public/assets.py [hunk 1]
```diff
@@ -252,3 +252,36 @@ def get_dag_asset_queued_events(
...
+    query = (
+        select(AssetDagRunQueue)
+        .join(AssetModel, AssetDagRunQueue.asset_id == AssetModel.id)
+        .where(*where_clause)
+    )
+    adrq = session.scalar(query)
+    if not adrq:
+        raise HTTPException(
+            status.HTTP_404_NOT_FOUND,
+            f"Queue event with dag_id: `{dag_id}` and asset uri: `{uri}` was not found",
```
Comment (Lee-W) [inline airflow/api_fastapi/core_api/routes/public/assets.py]:
```suggestion
            f"Queued event with dag_id: `{dag_id}` and asset uri: `{uri}` was not found",
```

File: tests/api_fastapi/core_api/routes/public/test_assets.py [hunk 2]
```diff
@@ -523,6 +523,42 @@ def test_should_respond_404(self, test_client):
...
+    def test_should_respond_200(self, test_client, session, create_dummy_dag):
+        dag, _ = create_dummy_dag()
+        dag_id = dag.dag_id
+        self.create_assets(session=session, num=1)
+        asset_id = 1
+        self._create_asset_dag_run_queues(dag_id, asset_id, session)
+        asset_uri = "s3://bucket/key/1"
+
+        response = test_client.get(
+            f"/public/dags/{dag_id}/assets/queuedEvent/{asset_uri}",
```
Comment (Lee-W) [inline tests/api_fastapi/core_api/routes/public/test_assets.py]: not related to this PR. We should keep this PR as it is.


but would like to confirm wit h @uranusjr, we probably would like to use the name for airflow 3?

File: tests/api_fastapi/core_api/routes/public/test_assets.py [hunk 3]
```diff
@@ -523,6 +523,42 @@ def test_should_respond_404(self, test_client):
...
+        asset_uri = "not_exists"
+
+        response = test_client.get(
+            f"/public/dags/{dag_id}/assets/queuedEvent/{asset_uri}",
+        )
+
+        assert response.status_code == 404
+        assert (
+            response.json()["detail"]
+            == "Queue event with dag_id: `not_exists` and asset uri: `not_exists` was not found"
```
Comment (Lee-W) [inline tests/api_fastapi/core_api/routes/public/test_assets.py]:
```suggestion
            == "Queued event with dag_id: `not_exists` and asset uri: `not_exists` was not found"
```

**Review by Lee-W — APPROVED** [review]
left one nitpick. but overall looks good

File: airflow/api_fastapi/core_api/routes/public/assets.py [hunk 4]
```diff
@@ -252,3 +252,36 @@ def get_dag_asset_queued_events(
...
+    adrq = session.scalar(query)
+    if not adrq:
+        raise HTTPException(
+            status.HTTP_404_NOT_FOUND,
+            f"Queued event with dag_id: `{dag_id}` and asset uri: `{uri}` was not found",
+        )
+
+    queued_event = QueuedEventResponse(created_at=adrq.created_at, dag_id=adrq.target_dag_id, uri=uri)
+
+    return QueuedEventResponse.model_validate(queued_event, from_attributes=True)
```
Comment (pierrejeambrun) [inline airflow/api_fastapi/core_api/routes/public/assets.py]:
```suggestion
    return QueuedEventResponse(created_at=adrq.created_at, dag_id=adrq.target_dag_id, uri=uri)
```

Validation is run on the constructor already.

**Review by pierrejeambrun — APPROVED** [review]
Great, just one small update required, feel free to merge after

File: airflow/api_fastapi/core_api/routes/public/assets.py (hunk 4 above)
Comment (amoghrajesh) [reply airflow/api_fastapi/core_api/routes/public/assets.py]: Oh yes, this is good

File: tests/api_fastapi/core_api/routes/public/test_assets.py (hunk 2 above)
Comment (amoghrajesh) [reply tests/api_fastapi/core_api/routes/public/test_assets.py]: @Lee-W resolving this PR for the sake of merging it, please create an issue if this is needed and you can assign it to me, we can prioritise and work on it

CC @pierrejeambrun @dstandish

**Pull request merged**